nixos/modules/security/acme.nix

```diff
@@ -332,11 +338,18 @@ in
                     ExecStart = pkgs.writeScript "acme-start" ''
                       #!${pkgs.runtimeShell} -e
                       test -L ${spath}/accounts -o -d ${spath}/accounts || ln -s ../accounts ${spath}/accounts
-                      ${pkgs.lego}/bin/lego ${renewOpts} || ${pkgs.lego}/bin/lego ${runOpts}
+                      LEGO_ARGS=(${runOpts})
+                      if [ -e ${spath}/certificates/${keyName}.crt ]; then
+                        REQUESTED_DOMAINS="${requestedDomains}"
+                        EXISTING_DOMAINS="$(openssl x509 -text -in ${spath}/certificates/${keyName}.crt -noout | perl -l -0777 -ne '@names=/\bDNS:([^\s,]+)/g; print join(";", sort @names);')"
```

Comment:
Maybe we can get rid of the perl dependency and instead rely on `coreutils` and `gnused`. I dabbled a bit and got to this:

```
openssl x509 -in cert.pem -noout -ext subjectAltName | tail -n1 | tr -d ' ' | sed -e 's/DNS://g' -e 's/,/;/g'
```

The SAN in the certificates are already sorted, probably because they were requested in that order.

Reply:
Great, I can also use the comma separator directly in `REQUESTED_DOMAINS` and use an even more simplified command (I also remove the whitespace in the sed command):
```
openssl x509 -in cert.pem -noout -ext subjectAltName | tail -n1 | sed -e 's/ *DNS://g'
```

Reply:
What is best/recommended when applying these changes in this repo:

- submit a new PR with a single clean commit
- edit this same PR

Reply:
Edit the same pr, you can squash your commits and force push them.

Reply:
done